src/wechaty.ts

```diff
@@ -521,6 +568,51 @@ export class Wechaty extends EventEmitter implements Sayable {
     return this
   }
 
+  /**
+   * @param   {WechatyPlugin | WechatyPlugin[]} plugins      - The plugins you want to use
+   *
+   * @return  {Wechaty}                                      - this for chaining,
+   *
+   * @desc
+   * For wechaty ecosystem, allow user to define a 3rd party plugin for the current wechaty instance.
+   *
+   * @example
+   * // Report all chat message to my server.
+   *
+   * Class WechatyReportPlugin {
+   *   constructor(options) {
+   *     this.options = options;
+   *   }
+   *   install(bot) {
+   *     bot.on('message', message => http.post(this.options.url, { data: message }))
+   *   }
+   * }
+   *
+   * bot.use(new WechatyReportPlugin({ url: 'http://somewhere.to.report.your.data.com' });
+   */
+  public use (plugins: WechatyPlugin | WechatyPlugin[]) {
+    ([] as WechatyPlugin[]).concat(plugins).forEach(plugin => {
+      if (!this.plugins.includes(plugin)) {
+        this.plugins.push(plugin)
+        this.installPlugin(plugin)
+      }
+    })
+    return this
+  }
+
+  private installGloablPlugin () {
+    Wechaty.globalPlugins.forEach(plugin => this.installPlugin(plugin))
+  }
+
+  private installPlugin (plugin: WechatyPlugin) {
+    if (plugin.installed) {
+      return
```

Comment:
About install one plugin multiple times: I believe it should be the responsibility of the user who is setting the plugin, to make sure everything is ok.

If we skip one plugin in silence for some reason, I believe it will introduce more puzzles for some use-cases.

Reply:
ok, since we removed the wehcaty.plugins property. we won't know they install twice. it's not a problem any more.